### ISSUE TITLE
[ parse, badMsg ] errors about implementation and interface

### ISSUE THREAD

**Issue by alissa-tung** [idris-lang/Idris2#1563]
# Steps to Reproduce

```idris
interface R (a, b : Type) where

A, B : Type

R A B where

t : Type

{x : _} -> {y : _} -> {z : _} -> (R x y, R y z) => R x z where
```
is **okay**, but

# Observed Behavior

```idris
interface R (a, b : Type) where

A, B : Type

R A B where

-- t : Type

{x : _} -> {y : _} -> {z : _} -> (R x y, R y z) => R x z where
```

and

```idris
interface R (a, b : Type) where

A, B : Type

R A B where

t : Type

{x, y : _} -> {z : _} -> (R x y, R y z) => R x z where
```

is not

**Comment by gallais** [comment]
Looks like it's probably the same kind of issue as https://github.com/idris-lang/Idris2/issues/1544#issuecomment-861447942

**Comment by alissa-tung** [comment]
closed for same as #1544
